### ISSUE TITLE
Adding a node to a cluster manually in Rancher 2?

### ISSUE THREAD

**Issue by mscreenie** [rancher/rancher#12562]
rancher/server: Rancher 2.0 beta2
kubernetes (if applicable): 1.10.0

I have a fairly typical cluster with 6 nodes in DigitalOcean created with Rancher 2.0. I have a new Linux Ubuntu VM outside of DigitalOcean in our corp network with an external IP. I want to add it to the cluster and have networking (Flannel). Previously I was able to do this with Version 1. 

Eventually the corporate node will have persistent storage with either NFS or Gluster and MySQL with sensitive data. I can't seem to do this with Rancher's UI except to add the VM in a separate cluster and AFAIK the networking is not shared between two clusters.

**Comment by vincent99** [comment]
You can't do it because the is no secure way to do this with any of the supported network plugins in k8s without a VPN. They are layer-2 or have no (stable) encryption support.

**Comment by mscreenie** [comment]
Thank you. I had this setup with Rancher version 1. 

The node also had a cluster IP setup by rancher, I'm not sure if communication was encrypted.

Is there any scope in Rancher 2 of adding custom nodes to clusters? In this case there would be no requirement to scale.

**Comment by gelinger777** [comment]
@vincent99  and what to do in case where we have created a cluster  in Hetzner Cloud and in the same datacenter we have a dedicated server, which we want to join the existing cluster?

**Comment by eleaner** [comment]
Hi everyone, I am fairly new to rancher and struggle with something that might be the same issue as discussed here.
I have a bunch of VPS in different Data centres
All the VPS are connected via wireguard mesh
I am trying to build rancher 2.0 cluster over wireguard and somehow ignore public addresses.
It does not seem to be working
some part of rancher insists on using public addresses to communicate
any idea? is it possible at all with rancher or do I have to try manual setup?

**Comment by eleaner** [comment]
@gelinger777 did you manage to find a solution?

**Comment by gelinger777** [comment]
@eleaner  yes. When you view cluster in API mode digg in the json, you will find there ready docker command .  You just need to run that docker command on the machine, and it will join the cluster. Would be good if you make sure that the docker version running on that machine matches the same version on the cluster.

**Comment by cjellick** [comment]
@eleaner is the rancher url you set at the beginning of setting up rancher on a public or private IP? Should be also able to find it on the settings page

**Comment by eleaner** [comment]
thank you guys, but I am not sure I understand your messages
my vpses sit on wireguard mesh
my rancher IP is one of the wireguard mesh
when adding a node I add wiregurad IP as public and internal

all because I want rancher to work on the wireguard mesh and do not open any other port
yet, it does not seem to work
wireguard finds the VPS public IP anyway and insists on using it.
I tried different cin drivers
I tried specifying interface in cluster yaml
nothing made it any better

in the end, I see the pods controlling CIN always use VPS always use VPS external IP
